tests/glsl-intrinsic/shader-subgroup/shader-subgroup-ballot.slang

```diff
@@ -0,0 +1,127 @@
+
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+precision highp float;
+precision highp int;
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=4):out,name=outputBuffer
+buffer MyBlockName2 
+{
+    uint data[];
+} outputBuffer;
+
+layout(local_size_x = 32) in;
+
+
+
+// required since at the time of making, __BuiltinType does not allow simple casting, so we have a split test
+
+__generic<T : __BuiltinLogicalType>
+bool test1BroadcastX() {
+    return true
+        && subgroupBroadcast(T(1), 0) == T(1)
+        && subgroupBroadcastFirst(T(1)) == T(1)
+        ;
+}
+__generic<T : __BuiltinLogicalType, let N : int>
+bool testVBroadcastX() {
+    typealias gvec = vector<T, N>;
+
+    return true
+        && subgroupBroadcast(gvec(T(1)), 0) == gvec(T(1))
+        && subgroupBroadcastFirst(gvec(T(1))) == gvec(T(1))
+        ;
+}
+
+__generic<T : __BuiltinFloatingPointType>
+bool test1BroadcastX() {
+    return true
+        && subgroupBroadcast(T(1), 0) == T(1)
+        && subgroupBroadcastFirst(T(1)) == T(1)
+        ;
+}
+__generic<T : __BuiltinFloatingPointType, let N : int>
+bool testVBroadcastX() {
+    typealias gvec = vector<T, N>;
+
+    return true
+        && subgroupBroadcast(gvec(T(1)), 0) == gvec(T(1))
+        && subgroupBroadcastFirst(gvec(T(1))) == gvec(T(1))
+        ;
+}
+bool testBroadcastX() {
+    return true
+        && test1BroadcastX<float>()
+        && testVBroadcastX<float, 2>()
+        && testVBroadcastX<float, 3>()
+        && testVBroadcastX<float, 4>()
+        //&& test1BroadcastX<double>() // hang
+        //&& testVBroadcastX<double, 2>()
+        //&& testVBroadcastX<double, 3>()
+        //&& testVBroadcastX<double, 4>()
+        //&& test1BroadcastX<half>() // shader_subgroup_preamble --> __requireGLSLExtension fails
+        //&& testVBroadcastX<half, 2>()
+        //&& testVBroadcastX<half, 3>()
+        //&& testVBroadcastX<half, 4>()
+        && test1BroadcastX<int>()
+        && testVBroadcastX<int, 2>()
+        && testVBroadcastX<int, 3>()
+        && testVBroadcastX<int, 4>()
+        //&& test1BroadcastX<int8_t>() // shader_subgroup_preamble --> __requireGLSLExtension fails
+        //&& testVBroadcastX<int8_t, 2>()
+        //&& testVBroadcastX<int8_t, 3>()
+        //&& testVBroadcastX<int8_t, 4>()
+        //&& test1BroadcastX<int16_t>() // shader_subgroup_preamble --> __requireGLSLExtension fails
+        //&& testVBroadcastX<int16_t, 2>()
+        //&& testVBroadcastX<int16_t, 3>()
+        //&& testVBroadcastX<int16_t, 4>()
+        //&& test1BroadcastX<int64_t>() // shader_subgroup_preamble --> __requireGLSLExtension fails
+        //&& testVBroadcastX<int64_t, 2>()
+        //&& testVBroadcastX<int64_t, 3>()
+        //&& testVBroadcastX<int64_t, 4>()
```

Comment:
Same here.
Let's limit the types to what the spec document requires

Reply:
(same answer as other comments)